daffodil-cli/src/main/scala/org/apache/daffodil/cli/Main.scala

```diff
@@ -617,7 +629,7 @@ class CLIConf(arguments: Array[String], stdout: PrintStream, stderr: PrintStream
   // Test Subcommand Options
   object test extends scallop.Subcommand("test") {
     banner(
-      """|Usage: daffodil test [-I <implementation>] [-l] [-r] [-i] <tdmlfile> [testnames...]
+      """|Usage: daffodil test [-I <implementation>] [-l] [-r] [-i] [-d <file> | -d -- | -t] <tdmlfile> [testnames...]
```

Comment:
`-d --` isn't really the right way to signify the ability to use `--`, since it sort of implies `--` is an argument to the `-d` option. But that's not really case. The `--` can go anywhere and just says that everything following it should not be treated as an option or an option argument, and should instead be treated as a trailing argument.

We happen to only really need it for the -d option since there's sometimes an ambiguity if the thing following it is the file with debug commands or a trailing argument. This ambiguity is whys why it's needed in some of our tests that do something like this:

```
daffodil test -d foo.tdml`
```
In this case you do need the `--` to say that foo.tdml is not actually an argument to the -d option, e.g.
```
daffodil test -d -- foo.tdml
```

But note that `--` isn't always necessary if there is no ambiguity. For example, if you do this
```
daffodil test -d -r foo.tdml 'test_foo_.*'
```
Then scallop knows that `-r` is an option so it can infer the `-d` option did not have an argument, and so the `--` isn't needed at all. This is also why you don't need `-d --` for the parse/unparse tests. In those tests there is no ambiguity because `-d` is immediately followed by another option.

But note that the same problem can still happen for the parse/unparse commands, it's not specific to the test command. For example, if you did this:
```
daffodil parse -s foo.dfdl.xsd -d input.bin
```
Scallop would think "input.bin" is the file of debug commands, there would be no trailing args, and so daffodil would parse stdin. Instead, you would need to do: 

```
daffodil parse -s foo.dfdl.xsd -d -- input.bin
```

The `--` is also useful in rare edge cases where the file you want to parse looks like an argument. For example, if you had a file that started with a hyphen you would also need to use `--`, e.g.
```
daffodil parse -s foo.dfdl.xsd -- --file_that_looks_like_an_argument.bin
```
(there's good reason we don't start files with hyphens).

So all this to say that `--` isn't really specific to `-d` so shouldn't be added to the usage like that. If we really want to make it clear that it can be used, probably the best way to do it is to add `[--]` before the trailing arguments. That would imply that `--` is optional, but if you want it to deal with ambiguity then put it before trailing arguments.

Git does something similar for commands like `git add`, and the man page even says:
```
       --
           This option can be used to separate command-line options from the list of files, (useful when filenames might be mistaken for command-line options).
```

I'm not sure if scallop would let us add that to the option description though.

Reply:
I decided to add the following comment to all the banners

`-- can be used to separate command-line options from trailing arguments`